From 856a9e0c102c509a797134785f366be2b20cdbed Mon Sep 17 00:00:00 2001
From: Aleksey Makarov <amakarov@marvell.com>
Date: Tue, 27 Nov 2018 22:50:58 +0300
Subject: [PATCH 0761/1051] net: octeontx2: Fix number of allocated send queue
 entries

sq->tso_hdrs and sq->sg arrays contain exactly qset->sqe_cnt
entries.  Allocate memory for exactly this number of entries.

Signed-off-by: Aleksey Makarov <amakarov@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/otx2_common.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_common.c b/drivers/net/ethernet/marvell/octeontx2/otx2_common.c
index 7bc1110a99ae..bb041475d351 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_common.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_common.c
@@ -543,15 +543,14 @@ static int otx2_sq_init(struct otx2_nic *pfvf, u16 qidx, u16 sqb_aura)
 		return err;
 
 	if (!pfvf->hw.hw_tso) {
-		err = qmem_alloc(pfvf->dev, &sq->tso_hdrs,
-				 qset->sqe_cnt + 1, TSO_HEADER_SIZE);
+		err = qmem_alloc(pfvf->dev, &sq->tso_hdrs, qset->sqe_cnt,
+				 TSO_HEADER_SIZE);
 		if (err)
 			return err;
 	}
 
 	sq->sqe_base = sq->sqe->base;
-	sq->sg = kcalloc((qset->sqe_cnt + 1),
-			 sizeof(struct sg_list), GFP_KERNEL);
+	sq->sg = kcalloc(qset->sqe_cnt, sizeof(struct sg_list), GFP_KERNEL);
 	if (!sq->sg)
 		return -ENOMEM;
 
-- 
2.17.1

